deltachat-ios/AppDelegate.swift

```diff
@@ -445,24 +450,54 @@ class AppDelegate: UIResponder, UIApplicationDelegate, UNUserNotificationCenterD
 
     // MARK: - misc.
 
-    func openDatabase() {
-        guard let databaseLocation = DatabaseHelper(dcContext: dcContext).updateDatabaseLocation() else {
+    func migrateToDcAccounts() {
+        let dcContext = DcContext()
+
+        // first database migration to shared container, can be removed at some point
+        // implemented in April 2020 (https://github.com/deltachat/deltachat-ios/pull/612)
+        let databaseHelper = DatabaseHelper(dcLogger: dcContext.logger)
+        guard let databaseLocation = databaseHelper.updateDatabaseLocation() else {
             fatalError("Database could not be opened")
         }
-        logger.info("open: \(databaseLocation)")
         dcContext.openDatabase(dbFile: databaseLocation)
+        if dcContext.isConfigured() {
+            dcContext.closeDatabase()
+            if dcAccounts.migrateAccount(dbLocation: databaseLocation) == 0 {
+                fatalError("Account could not be migrated")
+                // TODO: show error message in UI
+            }
+            databaseHelper.clearUnmanagedAccountData()
+        }
+    }
+
+    func reloadDcContext() {
+        locationManager.reloadDcContext()
+        notificationManager.reloadDcContext()
+        if dcAccounts.get().isConfigured() {
+            appCoordinator.presentTabBarController()
+        } else {
+            appCoordinator.presentWelcomeController()
+        }
     }
 
+    func openDatabase() {
```

Comment:
i suggest to take the chance to strike these tiny one-line-helper. the code is easier to read if you see eg. `dcAccounts.openDatabase()` directly.